test/test_models.py

```diff
@@ -69,16 +91,255 @@ def check_script(self, model, name):
             msg = str(e) + str(tb)
         self.assertTrue(scriptable, msg)
 
-    def _test_classification_model(self, name, input_shape):
-        # passing num_class equal to a number other than 1000 helps in making the test
-        # more enforcing in nature
-        model = models.__dict__[name](num_classes=50)
-        self.check_script(model, name)
-        model.eval()
-        x = torch.rand(input_shape)
-        out = model(x)
-        self.assertEqual(out.shape[-1], 50)
+    def _check_scriptable(self, model, expected):
+        if expected is None:  # we don't check scriptability for all models
+            return
+
+        actual = True
+        msg = ''
+        try:
+            torch.jit.script(model)
+        except Exception as e:
+            tb = traceback.format_exc()
+            actual = False
+            msg = str(e) + str(tb)
+        self.assertEqual(actual, expected, msg)
+
+
+class ClassificationCoverageTester(TestCase):
+
+    # Find all models exposed by torchvision.models factory methods (with assumptions)
+    def get_available_classification_models(self):
```

Comment:
Why are we only testing classification models for this ?

Reply:
The other model types are still using the old, implicit method of test generation.

Reply:
I don't think there should be two separate pathways at one time.

Reply:
I agree, but my brief was to cover the classification models. Also, I think the four other major classes of model should probably get broken out into separate files.